[BUG] compilation error with buffer_cache.h (indirectly through ollama pulling ml-explore/mlx-c)

**Describe the bug**

Attempting to compile MLX support with ollama on archlinux with gcc version `15.2.1 20260103` results in
```c++
In file included from ollama/build/_deps/mlx-src/mlx/backend/cuda/allocator.h:6,
                 from ollama/build/_deps/mlx-src/mlx/backend/cuda/allocator.cpp:3:
ollama/build/_deps/mlx-src/mlx/backend/common/buffer_cache.h: In member function ‘int mlx::core::BufferCache<T>::release_cached_buffers(size_t)’:
ollama/build/_deps/mlx-src/mlx/backend/common/buffer_cache.h:73:24: error: ‘find_if’ is not a member of ‘std’ [-Wtemplate-body]
   73 |         auto it = std::find_if(its.first, its.second, [this](const auto& el) {
      |                        ^~~~~~~
[288/878] Building CUDA object _deps/mlx-build/CMakeFiles/mlx.dir/mlx/backend/cuda/binary_two.cu.o
```

This seems to be because `std::find_if` is defined in `<algorithm>` but [buffer_cache.h](https://github.com/ml-explore/mlx/blob/main/mlx/backend/common/buffer_cache.h) does not seem to include it.

Since this is a dependency of a dependency, there doesn't seem to be an easy way to patch this.


**To Reproduce**

Easiest way is to run `g++ -std=gnu++17 mlx/backend/common/buffer_cache.h`, which should result in
```c++
mlx/backend/common/buffer_cache.h: In member function ‘int mlx::core::BufferCache<T>::release_cached_buffers(size_t)’:
mlx/backend/common/buffer_cache.h:73:24: error: ‘find_if’ is not a member of ‘std’ [-Wtemplate-body]
   73 |         auto it = std::find_if(its.first, its.second, [this](const auto& el) {
      |                        ^~~~~~~
```

**Expected behavior**
A clear and concise description of what you expected to happen.

**Desktop (please complete the following information):**
 - OS Version: Archlinux
 - Compiler: gcc 15.2.1

**Additional context**

This occured when building ollama using `cmake ... -DMLX_ENGINE=ON`, and ollama's cmake pulls v0.4.1 of mlx-c, which in turns pulls v0.30.1 of mlx.

--- a/mlx/backend/common/buffer_cache.h
+++ b/mlx/backend/common/buffer_cache.h
@@ -2,6 +2,7 @@
 
 #pragma once
 
+#include <algorithm>
 #include <cassert>
 #include <functional>
 #include <map>

